docs/sources/enterprise/saml/troubleshoot-saml.md

```diff
@@ -26,6 +26,42 @@ We only support one private key format: PKCS#8.
 
 The keys may be in a different format (PKCS#1 or PKCS#12); in that case, it may be necessary to convert the private key format.
 
+This creates a new pkcs8 key file.
+
 ```bash
-$ openssl pkcs8 -topk8 -nocrypt -in <yourkey> -out private.pem
+$ openssl req -x509 -newkey rsa:4096 -keyout key.pem -out cert.pem -days 365 -nodes​
+```
+
+#### **Convert** the private key format to base64
+
+This converts already created keys to base64
+
+Base64-encode the cert.pem and key.pem files:
+(-w0 switch is not needed on Mac, only for Linux)
+
+```sh
+$ base64 -w0 key.pem > key.pem.base64
+$ base64 -w0 cert.pem > cert.pem.base64
+```
+
+The base64-encoded values (`key.pem.base64, cert.pem.base64` files) are then used for certificate and private_key.
+
+The keys you provide should look like:
+
+It should look like:
+
+```
+-----BEGIN PRIVATE KEY-----
+...
+...
+-----END PRIVATE KEY-----
+```
+
+If you have a key that looks like:
```

Comment:
This confuses me. If I have a key that looks like the sample you provide, then what? The 'If' makes me feel like there is something missing here.

Reply:
Hmm il have a look at how I could rephrase this better